### ISSUE TITLE
[Cell Guide] Return 404 for Cell Type Page We Don't Have

### ISSUE THREAD

**Issue by tihuan** [chanzuckerberg/single-cell-data-portal#5632]
Problem:

I found someone using a Python script to parse all of our cell guide pages, which hit many cell type ids that don't have corresponding pages. This is causing a spike in 5xx errors in CloudFront

For example, one request was for https://cellxgene.cziscience.com/cellguide/CL_0035675, which we don’t have such cell type page

Solution:

We should return 404 instead of 500

Corresponding CloudFront Log file: https://s3.console.aws.amazon.com/s3/object/hosted-cellxgene-prod-hosted-cellxgene-cloudfront-logs?region=us-west-2&prefix=cloudfront/E112SLB7W4VBTS.2023-08-28-22.de9ddb7e.gz

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbVJHIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--698da3f82f32ce5d43fba7cf5db5032f7eb62479/Screenshot%202023-08-28%20at%203.23.33%20PM.png)

**Comment by tihuan** [comment]
CC: @atarashansky

**Comment by tihuan** [comment]
I think this is P1, since it's affecting prod level error report, so hopefully can be fixed this week. @niknak33 @atarashansky  what do you think? Thank you!

**Comment by niknak33** [comment]
Yeah, it's a P1. Would be nice to get a 404, I am sure the Google Indexing site will throw it up as well. @tihuan @atarashansky

**Comment by tihuan** [comment]
Awesome thanks, @niknak33 🙏 ! 

@atarashansky do you have the capacity to work on it this week or should we find someone else? Thank you!

**Comment by atarashansky** [comment]
The real question is how to prevent people from scraping CellGuide, lol. Would scrapers worsen the performance of CellGuide?

**Comment by atarashansky** [comment]
https://aws.amazon.com/waf/

**Comment by tihuan** [comment]
We still need to return the correct error code for SEO too!

**Comment by tihuan** [comment]
And also fixing this will stop triggering CloudWatch alert, since 5xx is for unexpected errors and this should be 404

**Comment by atarashansky** [comment]
Putting this on hold until after attending office hours with the security team about AWS WAF

**Comment by tihuan** [comment]
I might be missing something here, but I'm thinking we'd still need to return 404 even if we had WAF, right?
